### PR TITLE
Add HDF5 and torch support

### PR DESCRIPTION
**Pull request by balbok0**

### CODE REVIEW

File: src/data_utils/hdf5_processing.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            # We need the `pose` because `twist` is given in the world frame,
+            # while we want it in the robot base frame.
+            linear_vel = np.cos(theta) * twist[:, 0] + np.sin(theta) * twist[:, 1]
+            # Transverse vel should be zero except for projection errors and
+            # times when the robot is slipping sideways.
+            transverse_vel = -np.sin(theta) * twist[:, 0] + np.cos(theta) * twist[:, 1]
+            return np.array([linear_vel, transverse_vel, theta_dot])
+
+        p_pose = planar_pose(h5_file["input_pair/state/pose/orientation"][:]).T
+        p_twist = planar_twist(h5_file["input_pair/state/twist/linear"][:], p_pose[:, 2], h5_file["input_pair/state/twist/angular"][:, 2])
```
Comment (balbok0) [inline src/data_utils/hdf5_processing.py:72]: I know these shouldn't be ground truth, but hdf doesn't log GT yet. I will make a PR for that.

File: src/train.py [hunk 2]
```diff
@@ -106,15 +108,16 @@
...
+        uni_model = UnicycleModel(D=D, H=H, P=P, delay_steps=1)
+        uni_model.train(train_set, n_steps=N_TRAIN_STEPS)
+        _, _, umx, umy = uni_model.compare_qualitative(
+                viz_seq, start_idx=viz_start_idx, n_steps=n_steps)
+        line, = plt.plot(umx, umy, color='red', label='Unicycle model')
+        uni_score = uni_model.evaluate(test_set, n_steps=N_EVAL_STEPS)
+        print("Unicycle model:", uni_score)
 
-    linear_model = LinearModel(D=D, H=H, P=P, delay_steps=1)
+    linear_model = LinearModel(D=D, H=H, P=P, delay_steps=1, lambda_=1e-2)
```
Comment (balbok0) [inline src/train.py:122]: I chose 1e-2 arbitrarily. It is not tuned.

**Comment by balbok0** [comment]
Updated PR with torch implementation of `compare_qualitative`. A simple model with 2 hidden layers (each of size 16) and sigmoid function between them outperforms the linear model:

Linear Model:
Linear Model

Torch Model:
Torch Model
